terraform/modules/front_door/security_groups.tf

```diff
@@ -0,0 +1,55 @@
+resource "aws_security_group" "load_balancer" {
+  name        = "${var.prefix}-load-balancer-security-group"
+  description = "Load Balancer security group"
+  vpc_id      = var.vpc_id
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_vpc_security_group_ingress_rule" "load_balancer_http_ingress" {
+  #checkov:skip=CKV_AWS_260:ingress from all IPs to port 80 required as load balancer is public
+  description       = "Allow ingress on port 80 from all IP addresses"
+  cidr_ipv4         = "0.0.0.0/0"
+  ip_protocol       = "tcp"
+  from_port         = 80
+  to_port           = 80
+  security_group_id = aws_security_group.load_balancer.id
+}
+
+resource "aws_vpc_security_group_ingress_rule" "load_balancer_https_ingress" {
+  description       = "Allow ingress on port 443 from all IP addresses"
+  cidr_ipv4         = "0.0.0.0/0"
+  ip_protocol       = "tcp"
+  from_port         = 443
+  to_port           = 443
+  security_group_id = aws_security_group.load_balancer.id
+}
+
+resource "aws_vpc_security_group_egress_rule" "load_balancer_http_egress" {

```

Comment:
These two should actually not be necessary - what AWS suggests for an application load balancer outbound is just the instance security group on the instance listener port and health check port (if different).

Note that responses to inbound traffic should always be allowed, regardless of the outbound rules.